### PR TITLE
FEATURE: RAIL-586 - Advanced examples of UI SDK drills + RELATED: FET-58

### PR DESCRIPTION
**Pull request by tomasreichmann**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1312/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1076/ : SUCCESS in 4m 39s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/549/ : SUCCESS in 3m 24s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1320/ : SUCCESS in 21s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1086/ : SUCCESS in 1m 30s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/559/ : SUCCESS in 3m 03s

File: examples/src/components/DrillWithExternalDataExample.jsx [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+            name,
+            uri: `${locationStateAttributeUri}/elements?id=${id}`
+        };
+        this.setState({
+            state,
+            location: null
+        });
+    }
+
+    onClearState = () => {
```
Comment (lhorak) [inline examples/src/components/DrillWithExternalDataExample.jsx]: I'd keep the naming consistent, so if there are onLocationDrill and onStateDrill, I'd use onStateClear and onLocationClear or vice versa

File: examples/src/components/DrillWithExternalDataExample.jsx [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+            visualizationAttribute: {
+                localIdentifier: 'locationName',
+                displayForm: {
+                    identifier: locationNameDisplayFormIdentifier
+                }
+            }
+        };
+        const employeeTableFilters = [];
+        if (state) {
+            employeeTableFilters.push({
```
Comment (lhorak) [inline examples/src/components/DrillWithExternalDataExample.jsx]: I'd extract this into some kind of a function, such as addPositiveFilter(filterArray, dfIdentifier, elementUris), it's used multiple times through the code

File: examples/src/components/DrillWithExternalDataExample.jsx [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+                    }
+                },
+                alias: 'Total Sales'
+            }
+        };
+
+        const salesTableFilters = [];
+        if (state) {
+            salesTableFilters.push({
+                positiveAttributeFilter: {
```
Comment (lhorak) [inline examples/src/components/DrillWithExternalDataExample.jsx]: DRY

File: examples/src/components/DrillWithExternalDataExample.jsx [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+                    displayForm: {
+                        identifier: locationStateDisplayFormIdentifier
+                    },
+                    in: [state.uri]
+                }
+            });
+        }
+        if (location) {
+            salesTableFilters.push({
+                positiveAttributeFilter: {
```
Comment (lhorak) [inline examples/src/components/DrillWithExternalDataExample.jsx]: DRY

File: examples/src/components/DrillWithExternalDataExample.jsx [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+                    displayForm: {
+                        identifier: locationStateDisplayFormIdentifier
+                    },
+                    in: [state.uri]
+                }
+            });
+        }
+        if (location) {
+            employeeTableFilters.push({
+                positiveAttributeFilter: {
```
Comment (lhorak) [inline examples/src/components/DrillWithExternalDataExample.jsx]: DRY

File: examples/src/components/DrillWithExternalDataExample.jsx [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+                    displayForm: {
+                        identifier: locationNameDisplayFormIdentifier
+                    },
+                    in: [location.uri]
+                }
+            });
+        }
+
+        const totalSalesTable = (
```
Comment (lhorak) [inline examples/src/components/DrillWithExternalDataExample.jsx]: Why table if there is ColumnChart?

File: examples/src/components/DrillWithExternalDataExample.jsx [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+                        identifier: locationNameDisplayFormIdentifier
+                    }]}
+                    onFiredDrillEvent={this.onLocationDrill}
+                />
+            </div>
+        );
+
+        const { employee, employee3rdPartyData } = this.state;
+
+        let employeeDetails = null;
```
Comment (lhorak) [inline examples/src/components/DrillWithExternalDataExample.jsx]: I'd move this to separate render function which would return the component according to employee3rdPartyData:

```
renderEmployeeDetails(employee) {
  if (employee3rdPartyData.isError) {
           return <ErrorComponent height={150} message="There was an error getting employee details" />;
  } else if (employee3rdPartyData.isLoading) {
         return  <LoadingComponent height={150} />;
      } else if (employee3rdPartyData.data) {
         return <EmployeeProfile {...employee3rdPartyData.data} />;
      }

     return null;
}

File: examples/src/components/DrillWithExternalDataExample.jsx [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+                        Sales from {!state && !location && 'All Locations'}
+                        {state && (<span>
+                            {state.name}&nbsp;<button onClick={this.onClearState} className="button button-primary button-small button-icon-only icon-cross s-sales-heading-clear-state" />
+                        </span>)}
+                        {location && (<span>
+                            {location.name}&nbsp;<button onClick={this.onClearLocation} className="button button-primary button-small button-icon-only icon-cross s-sales-heading-clear-location" />
+                        </span>)}
+                    </h3>
+                    {totalSalesTable}
+                    {employee && (<div>
```
Comment (lhorak) [inline examples/src/components/DrillWithExternalDataExample.jsx]: `employee && renderEmployeeDetails(..)`

Comment (tomasreichmann) [reply examples/src/components/DrillWithExternalDataExample.jsx]: This is actually two stage render. First when employee is selected, we render the title, but employeeDetails is null, then we render the details when the 3rd party call is resolved. So I will not move it to renderEmployeeDetails.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1327/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1098/ : SUCCESS in 2m 29s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/571/ : SUCCESS in 2m 11s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/263/ : SUCCESS in 18s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1100/ : SUCCESS in 4m 03s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/439/ : UNSTABLE in 7m 54s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/573/ : SUCCESS in 3m 35s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/264/ : FAILURE in 24s
- gooddata-react-components-unit-tests-zuul-docker-el7 : SKIPPED
- gooddata-react-components-storybook-zuul-docker : SKIPPED
- gooddata-react-components-examples-validation-zuul-docker-el7 : SKIPPED

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/265/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1101/ : FAILURE in 3m 03s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/440/ : UNSTABLE in 8m 18s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/574/ : SUCCESS in 5m 10s

**Comment by zbycz** [comment]
lgtm

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1334/ : SUCCESS in 34s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1104/ : FAILURE in 3m 23s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/577/ : FAILURE in 3m 00s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/266/ : SUCCESS in 21s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1105/ : FAILURE in 3m 28s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/442/ : SUCCESS in 8m 02s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/578/ : SUCCESS in 4m 31s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1338/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/384/ : FAILURE in 5m 27s

**Comment by tomasreichmann** [comment]
extended test - storybook

**Comment by tomasreichmann** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1340/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1107/ : SUCCESS in 3m 17s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/580/ : SUCCESS in 3m 30s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/446/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1339/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/385/ : UNSTABLE in 9m 01s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/267/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1109/ : FAILURE in 2m 26s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/447/ : UNSTABLE in 7m 14s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/582/ : SUCCESS in 3m 08s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/268/ : SUCCESS in 20s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1111/ : SUCCESS in 1m 22s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/448/ : SUCCESS in 6m 35s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/584/ : SUCCESS in 2m 13s

**Pull request merged**